### ISSUE TITLE
OAuth login failed: OAuthAccountNotLinked

### ISSUE THREAD

**Issue by sebgeekinside** [hoarder-app/hoarder#736]
### Describe the Bug

It is not possible to sign in using the oauth provider Keycloak.
Sign in without using Keycloak is ok.

### Steps to Reproduce

My oauth provider: Keycloak 26.0.7 (docker)

Click Sign in with Keycloak redirect to Keycloak login page
Click Sign In redirect back to hoarder login page, but got error OAuth login failed: OAuthAccountNotLinked

### Expected Behaviour

Succeed in sign in using the oauth provider Keycloak.

### Screenshots or Additional Context

![image](https://github.com/user-attachments/assets/3e8e9585-a1a4-41fc-9c89-af89b01645b3)

### Exact Hoarder Version

0.19.0

**Comment by MohamedBassem** [comment]
If before setting up oauth, you had an account with the same email, you'll have to set `OAUTH_ALLOW_DANGEROUS_EMAIL_ACCOUNT_LINKING=true` in the env file.

**Comment by sebgeekinside** [comment]
Indeed, it works but how it is possible to create an account using directly the oauth provider ?
In order to not use `OAUTH_ALLOW_DANGEROUS_EMAIL_ACCOUNT_LINKING=true`.

**Comment by MohamedBassem** [comment]
you'll need to delete the account with the email you want to use and recreate it by logining with oauth. It's worth noting that it's fine to enable account linking if you trust tehe oauth provider which in this case you do. So I wouldn't worry too much about much about this variable.

**Comment by sebgeekinside** [comment]
Ok, thanks a lot for the answers!

**Comment by sebgeekinside** [comment]
See above the explanations.
